src/mip/bound_strengthening.jl

```diff
@@ -0,0 +1,66 @@
+function bound_strengthening!(ps::PresolveData{T}, j::Int, ϵ::T=eps(T), ϵ_int::T=eps(T)) where{T}
+    # Column was already removed or the variable is continous.
+    ps.colflag[j] || return nothing
+    (ps.var_types[j] == CONTINUOUS) && return nothing
+
+    for i in 1:ps.pb0.ncon
+        # Row was already removed.
+        ps.rowflag[i] || continue
+        row = ps.pb0.arows[i]
+        j_ind = findfirst(isequal(j), row.nzind)
+        if (j_ind == nothing)
+            row_j = 0
+        else
+            row_j = row.nzval[j_ind]
+        end
+        if (abs(row_j) <= ϵ) # a_{i,j} is 0.
+            continue
+        end
+        lrow = ps.lrow[i]
+        urow = ps.urow[i]
+        upper = calc_upper_bound_except_one(row, ps.lcol, ps.ucol, j)
+        lower = calc_lower_bound_except_one(row, ps.lcol, ps.ucol, j)
+        if (row_j > T(0))
+            if (urow - lower != NaN)
+                ps.ucol[j] = T(min(ps.ucol[j],
+                                approx_floor((urow - lower) / row_j, ϵ_int)))
```

Comment:
The functions approx_floor() and approx_ceil() are used in both bound strengthening and round integer bound. Should we write a separate file for those functions?

Reply:
Yes, please move it to the util.jl file.